diff --git a/src/TBMegaMenuBuilder.php b/src/TBMegaMenuBuilder.php
index f586428..fcf8a84 100644
--- a/src/TBMegaMenuBuilder.php
+++ b/src/TBMegaMenuBuilder.php
@@ -128,11 +128,7 @@ class TBMegaMenuBuilder implements TBMegaMenuBuilderInterface {
    * {@inheritdoc}
    */
   public function loadEntityBlock(string $block_id) {
-    $block = $this->entityTypeManager->getStorage('block')->load($block_id);
-    // Ensure the current user has permissions to view the block.
-    if ($block->access('view')) {
-      return $block;
-    }
+    return $this->entityTypeManager->getStorage('block')->load($block_id);
   }
 
   /**
@@ -247,10 +243,7 @@ class TBMegaMenuBuilder implements TBMegaMenuBuilderInterface {
       $entities = $block_storage->loadMultiple($entity_ids);
       $_blocks_array = [];
       foreach ($entities as $block_id => $block) {
-        // Ensure the current user has access to view the block and the block
-        // is not provided by the tb_megamenu module.
-        if ($block->get('settings')['provider'] != 'tb_megamenu'
-        && $block->access('view')) {
+        if ($block->get('settings')['provider'] != 'tb_megamenu') {
           $_blocks_array[$block_id] = $block->label();
         }
       }
